### PR TITLE
quartus-prime-lite: improvements & fixes

### PR DESCRIPTION
**Pull request by bjornfor**
## Description of changes

Best reviewed per commit:

- quartus-prime-lite: run mkdir as needed
- quartus-prime-lite: lower case local shell variables
- quartus-prime-lite: list progs to wrap in sh instead of Nix
- quartus-prime-lite: add /lib/ld-lsb*.so.3 dynamic loaders to FHS env
- quartus-prime-lite: add lmutil
- quartus-prime-lite: add vcom, vdel, vmap
- quartus-prime-lite: expose all of quartus/bin/*
- quartus-prime-lite: eliminate two unneeded execve syscalls
- quartus-prime-lite: don't overwrite LD_PRELOAD
- quartus-prime-lite: buildFHSEnvChroot -> buildFHSEnv

UPDATE: More commits:
- quartus-prime-lite: modelsim: fix compiling encrypted device models
- quartus-prime-lite: test building encrypted device model
- quartus-prime-lite: use runtimeShell in wrappers

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bjornfor** [comment]
Merging in two days unless there are objections.

**Review by Mic92 — APPROVED** [review]
Looks fine visually. Didn't test it.

File: pkgs/applications/editors/quartus-prime/default.nix [hunk 1]
```diff
@@ -47,46 +57,67 @@ in buildFHSEnvChroot rec {
...
+      "${unwrapped}"/quartus/sopc_builder/bin/qsys-{generate,edit,script}
+      # Should we install all executables?
+      "${unwrapped}"/modelsim_ase/bin/{vsim,vlog,vlib,vcom,vdel,vmap}
+      "${unwrapped}"/modelsim_ase/linuxaloem/lmutil
+    )
+
+    wrapper=$out/bin/${name}
+    progs_wrapped=()
+    for prog in ''${progs_to_wrap[@]}; do
+        relname="''${prog#"${unwrapped}/"}"
```
Comment (kwohlfahrt) [inline pkgs/applications/editors/quartus-prime/default.nix:76]: I'm not up on my bash-isms, this is doing something like `realpath --relative-to "${unwrapped}" "$prog"`?

**Review by kwohlfahrt — APPROVED** [review]
I don't have time to test, but this looks good to me.

**Comment by tpwrules** [comment]
I will test later today

**Comment by bjornfor** [comment]
> qsys-edit needs dejavu_fonts added to its requirements in order to launch and gnumake in order to generate HDL, that needs to be addressed so that works properly. Not a new problem but worth finally fixing.

I'm not familiar with that tool, but the GUI comes up (with seemingly working fonts) both on nixpkgs-unstable and on this PR branch. I suggest making further imrovements in other PRs.

Good to merge or does anyone want to test more?

**Comment by tpwrules** [comment]
> I'm not familiar with that tool, but the GUI comes up (with seemingly working fonts) both on nixpkgs-unstable and on this PR branch. I suggest making further imrovements in other PRs.

All this should need is to add the two packages under `# qsys requirements` in `targetPkgs`.

**Comment by bjornfor** [comment]
> > I'm not familiar with that tool, but the GUI comes up (with seemingly working fonts) both on nixpkgs-unstable and on this PR branch. I suggest making further imrovements in other PRs.
> 
> All this should need is to add the two packages under `# qsys requirements` in `targetPkgs`.

I see, but since I don't know how to test it, and qsys-edit already launch for me, I'd prefer if that can be done in another PR. (You also mentioned other fixes (qemu), which I don't feel comfortable with including here just because it's related to quartus.)

**Pull request merged**